app/src/main/java/com/github/se/wanderpals/ui/screens/trip/Map.kt

```diff
@@ -184,53 +212,33 @@ fun Map(tripId: String, mapViewModel: MapViewModel) {
         })
   }
 }
-// request the location of the address from the nominatim api
-fun requestLocation(
-    address: String,
-    onSuccessRequest: (List<String>, LatLng) -> Unit,
-    onFailureRequest: () -> Unit
-) {
-  val client = OkHttpClient()
-
-  val url =
-      "https://nominatim.openstreetmap.org/search?q=${
-            address.replace(
-                " ",
-                "%20"
-            )
-        }&format=json"
-
-  val request = Request.Builder().url(url).build()
 
-  client
-      .newCall(request)
-      .enqueue(
-          object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-              MainScope().launch { onFailureRequest() }
-            }
-
-            override fun onResponse(call: Call, response: Response) {
-              if (response.isSuccessful) {
-                try {
-                  val jsonArray = JSONArray(response.body?.string())
-                  if (jsonArray.length() > 0) {
-                    val jsonObject = jsonArray.getJSONObject(0)
-                    val lat = jsonObject.getDouble("lat")
-                    val lon = jsonObject.getDouble("lon")
-                    val displayName = jsonObject.getString("display_name").split(",")
-                    // store the latitude and longitude of the address
-                    val location = LatLng(lat, lon)
-                    MainScope().launch { onSuccessRequest(displayName, location) }
-                  } else {
-                    MainScope().launch { onFailureRequest() }
-                  }
-                } catch (e: Exception) {
-                  MainScope().launch { onFailureRequest() }
-                }
-              } else {
-                MainScope().launch { onFailureRequest() }
-              }
-            }
-          })
-}
+suspend fun getAddressPredictions(
+    client: PlacesClient,
+    sessionToken: AutocompleteSessionToken = AutocompleteSessionToken.newInstance(),
+    inputString: String,
+    location: LatLng? = null,
+    onSuccess: (List<AutocompletePrediction>) -> Unit = {},
+    onFailure: (Exception?) -> Unit = { throw Exception("Place not found") }
+) =
+    suspendCoroutine<Unit> {
+      val request =
+          FindAutocompletePredictionsRequest.builder()
+              .setLocationBias(
+                  location?.let { locationBias ->
+                    RectangularBounds.newInstance(
+                        LatLng(locationBias.latitude - 0.1, locationBias.longitude - 0.1),
+                        LatLng(locationBias.latitude + 0.1, locationBias.longitude + 0.1))
+                  })
+              .setOrigin(location)
+              .setCountries("CH")
```

Comment:
As discussed, it would be great to add the countries covered by the trips

Reply:
I have limited it to CH to reduce the cost of the query for now, of course it will become more precise later as needed